model/user/user.go

```diff
@@ -222,6 +222,33 @@ func (u *DBUser) AddFavoritedProject(identifier string) error {
 	return nil
 }
 
+func (u *DBUser) RemoveFavoriteProject(identifier string) error {
+	if !util.StringSliceContains(u.FavoriteProjects, identifier) {
+		return errors.Errorf("project '%s' does not exist in user's favorites", identifier)
+	}
+	before := u.FavoriteProjects
+
+	update := bson.M{
+		"$pull": bson.M{FavoriteProjectsKey: identifier},
+	}
+	if err := UpdateOne(bson.M{IdKey: u.Id}, update); err != nil {
```

Comment:
combine the update query into the update document.

Reply:
I kind of prefer this tbh because that will line break weirdly

Reply:
i also prefer this and it is how every other method in this file is written. I am not in the camp that removing every variable possible enhances readability. Variables are awesome because they tell me what the purpose of this code is.

Reply:
they reduce readability, and names can and often do drift, and so I read this and I think "the variable is declared seperate, I wonder if it's mutated before it's passed to the update", and then I have to hold that state while I read the function.

Reply:
that's where consts are helpful but it looks like inferred type declaration with `:=` is preferred in Go, at least in our code.

Reply:
i removed the variable @tychoish but it becomes far more unreadable like this: 
```
if err := UpdateOne(bson.M{IdKey: u.Id}, bson.M{
    "$pull": bson.M{FavoriteProjectsKey: identifier},
}); err != nil {
    return err
}
```